### ISSUE TITLE
excel crash on exit when calling xlwings from an addin

### ISSUE THREAD

**Issue by sdementen** [ZoomerAnalytics/xlwings#246]
I have moved the xlwings VBA code to an addin to which excel files refers to to use RunPython.

When I open a workbook that uses the addin and call the RunPython method, I will get the message ![error](https://cloud.githubusercontent.com/assets/1304950/10270169/f015e3b2-6ae9-11e5-9c24-24d7a644b7d0.gif). 
If I do not call the RunPython method, Excel will exit without error message.

However, if I move the VBA code within the workbook (and remove the reference to the xlwings addin) and I call the RunPython method, Excel exits cleanly.

Any hint on what could cause this strange behavior (that I did not get when using the VBA code from xlwings 0.3.0) ?

**Comment by fzumstein** [comment]
I personally haven't looked into the add-in topic yet but there's some discussion at #153. Not sure though that will help...

**Comment by sdementen** [comment]
Not really. I already converted in an addin the xlwings 3.0 version successfully.
It looks like a nasty bug (uncleaned reference ?).
I do not have it when using the OPTIMIZED_CONNECTION. Would it be possible to connect to xlwings with OPTIMIZED_CONNECTION and then close/exit the interpreter after having called the command ? something like (does not work)

        If OPTIMIZED_CONNECTION = True Then
            Py.SetAttr Py.module("xlwings._xlwindows"), "xl_workbook_current", ActiveWorkbook
            Py.Exec "" & PythonCommand & ""
            Py.Exec "" & "exit()" & ""
        Else

I have tried "import sys; sys.exit()" but it raises exceptions.

BTW, in OPTIMIZED_CONNECTION, how are the exceptions managed ? the Py.Exec line throws an error that could be handled with an "on error" but is it possible to access the attributes of the exception that is raised ? what are the functions available in Py (xlwings.dll) ? is the code of xlwings.dll published ?
